example/examplechecker.py

```diff
@@ -1,128 +1,197 @@
+                raise BrokenServiceException(
+                    "received invalid response on /profile endpoint"
+                )
         else:
             raise ValueError(
                 "Call_idx {} not supported!".format(self.flag_idx)
             )  # Internal error.
 
-    def putnoise(self):  # type: () -> None
+    def putnoise(self) -> None:
         """
         This method stores noise in the service. The noise should later be recoverable.
         The difference between noise and flag is, tht noise does not have to remain secret for other teams.
         This method can be called many times per round. Check how often using self.flag_idx.
         On error, raise an EnoException.
         :raises EnoException on error
-        :return this function can return a result if it wants
-                if nothing is returned, the service status is considered okay.
-                the preferred way to report errors in the service is by raising an appropriate enoexception
         """
-        self.team_db["noise"] = self.noise
+        credentials = self.generate_credentials()
+        self.team_db[self.flag] = credentials
+        self.register_and_login(credentials)
+
+        category = secrets.choice(
+            [
+                "Python",
+                "NodeJS",
+                "C",
+                "Rust",
+                "Go",
+                "C#",
+                "C++",
+                "Prolog",
+                "OCL",
+                "Julia",
+            ]
+        )
 
-    def getnoise(self):  # type: () -> None
+        # we are overwriting the credentials on purpose since we don't need them later in this case
+        self.team_db[self.noise] = category
+
+        res = self.http_post(
+            "/posts",
+            json={"content": self.noise, "category": category, "public": True},
+        )
+        assert_equals(res.status_code, 200)
+
+    def getnoise(self) -> None:
         """
         This method retrieves noise in the service.
         The noise to be retrieved is inside self.flag
         The difference between noise and flag is, tht noise does not have to remain secret for other teams.
         This method can be called many times per round. Check how often using flag_idx.
         On error, raise an EnoException.
         :raises EnoException on error
-        :return this function can return a result if it wants
-                if nothing is returned, the service status is considered okay.
-                the preferred way to report errors in the service is by raising an appropriate enoexception
         """
+        category = self.team_db[self.noise]
+
+        res = self.http_get("/posts", json={"category": category})
+        assert_equals(res.status_code, 200)
+
         try:
-            assert_equals(self.team_db["noise"], self.noise)
-        except KeyError:
-            raise BrokenServiceException("Noise not found!")
+            for post in res.json()["posts"]:
```

Comment:
Ich hatte mir bei json force=true angewöhnt, allerdings vergessen, warum...

Reply:
I don't think this actually does anything? According to https://requests.readthedocs.io/en/master/api/#requests.Response.json this is passed to `json.loads()`, but that only leads to
```
>> json.loads("{}", force=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.8/json/__init__.py", line 370, in loads
    return cls(**kw).decode(s)
TypeError: __init__() got an unexpected keyword argument 'force'
```

Reply:
https://flask.palletsprojects.com/en/1.1.x/api/#flask.Request.get_json <- this is probably what you meant